tests/Tests/500__jupyterhub/test-versions.robot

```diff
@@ -9,40 +9,69 @@ Suite Setup      Begin Web Test
-    Launch JupyterHub From RHODS Dashboard Link
-  ELSE
-    Launch JupyterHub From RHODS Dashboard Dropdown
-  END
-  Login To Jupyterhub  ${TEST_USER.USERNAME}  ${TEST_USER.PASSWORD}  ${TEST_USER.AUTH_TYPE}
-  ${authorization_required} =  Is Service Account Authorization Required
-  Run Keyword If  ${authorization_required}  Authorize jupyterhub service account
-  Wait Until Page Contains Element  xpath://span[@id='jupyterhub-logo']
-  Verify Libraries In Base Images
-
-*** Keywords ***
-Verify Libraries In Base Images
-    FOR  ${img}  IN  @{notebook_images}
-        @{list} =  Create List
-        ${text} =  Fetch Image Description Info  ${img}
-        Append To List  ${list}  ${text}
-        ${tmp} =  Fetch Image Tooltip Info  ${img}
-        ${list} =  Combine Lists  ${list}  ${tmp}
-        Log  ${list}
-        Spawn Notebook With Arguments  image=${img}
-        Check Versions In JupyterLab  ${list}
-        Clean Up Server
-        Stop JupyterLab Notebook Server
-        Go To  ${ODH_DASHBOARD_URL}
-        Wait for RHODS Dashboard to Load
+    [Tags]  ODS-695
+    Wait for RHODS Dashboard to Load
+    ${version-check} =  Is RHODS Version Greater Or Equal Than  1.4.0
+    IF  ${version-check}==True
         Launch JupyterHub From RHODS Dashboard Link
-        Wait Until JupyterHub Spawner Is Ready
+    ELSE
+        Launch JupyterHub From RHODS Dashboard Dropdown
     END
+    Login To Jupyterhub  ${TEST_USER.USERNAME}  ${TEST_USER.PASSWORD}  ${TEST_USER.AUTH_TYPE}
+    ${authorization_required} =  Is Service Account Authorization Required
+    Run Keyword If  ${authorization_required}  Authorize jupyterhub service account
+    Wait Until Page Contains Element  xpath://span[@id='jupyterhub-logo']
+
+Verify Libraries in Minimal Image
+    @{additional_libs} =    Create List
+    ${status} =  Verify Libraries In Base Image    s2i-minimal-notebook    ${additional_libs}
+    Append To List  ${status_list}  ${status}
+
+Verify Libraries in SDS Image
+    @{additional_libs} =    Create List
+    Append To List    ${additional_libs}    JupyterLab v3.2    Notebook v6.4
+    ${status} =  Verify Libraries In Base Image    s2i-generic-data-science-notebook    ${additional_libs}
+    Append To List  ${status_list}  ${status}
+
+Verify Libraries in PyTorch Image
+    [Tags]  ODS-215  ODS-216  ODS-217  ODS-218
+    @{additional_libs} =    Create List
+    Append To List    ${additional_libs}    JupyterLab v3.2    Notebook v6.4
+    ${status} =  Verify Libraries In Base Image    pytorch    ${additional_libs}
+    Append To List  ${status_list}  ${status}
+
+Verify Libraries in Tensorflow Image
+    [Tags]  ODS-204  ODS-205  ODS-206  ODS-207
+    @{additional_libs} =    Create List
+    Append To List    ${additional_libs}    JupyterLab v3.2    Notebook v6.4
+    ${status} =  Verify Libraries In Base Image    tensorflow    ${additional_libs}
+    Append To List  ${status_list}  ${status}
+
+Verify All Images And Spawner
```

Comment:
why creating a TC to check the results from the previous ones? Isn't possible to do the status check in each TC?

Reply:
Done like this to have the Polarion ID in a single TC; the test needs to check that all spawner descriptions match with all installed versions.

Reply:
hmm yeah I got it..an alternative maybe would have been to turn all the single TCs (e.g., `Verify Libraries in Tensorflow Image`) into keywords, and write `Verify All Images And Spawner` as a list of checks (e.g., `Run Keyword and Continue on Failure`). I'm saying this because otherwise we have in the logs many TCs (one for each image) performing the checks, and then an additional TC which repeats the results of the signle Tcs, without adding value.

But, just one opinion. I would still agree with the current structure. And if in the future we'll see it doesn't fit, we could change

Reply:
If I understood your suggestion correctly, I don't think it would work because we would end up doing all the checks (i.e. calling all the keywords) from one TC, which would bring us back to the previous problem I was trying to solve (all Polarion IDs linked to a single TC). If that's not what you meant, I think the other option would be to create different keywords and different TCs (where each TC only calls one keyword) but that's essentially the same as it is now, the only difference is that the "logic" is in the keyword rather than the TC. 

Let me know if I have misunderstood anything, otherwise I think this is the better approach for now

Reply:
yeah you got what I meant. But I thought the initial problem was to have the same Polarion ID assigned to multiple TCs. As of now we have different case where each TC has more than once Polarion IDs. 
IMO, the real solution to the problem is to review all the polarion TC to remove redundancy. 

With the current implementation here, I'm not sure about the sense of having the same check done in two different TCs. And you can have that when the check fails, the single-image TC (which only returns the status) is still SUCCESS, while the cumulative TC is FAILED. hmmm correct me if I'm wrong

Reply:
Fixed the last point in the newest commit